src/routes/paste.rs

```diff
@@ -195,12 +195,31 @@ pub async fn insert(
         .ok_or_else(|| StatusCode::UNSUPPORTED_MEDIA_TYPE.into_response())?;
 
     if content_type == headers::ContentType::form_url_encoded() {
+        let is_https = if let Some(header_host) = headers
+            .get(http::header::HOST)
+            .and_then(|val| val.to_str().ok())
+        {
+            headers.get(http::header::ORIGIN).is_some_and(|value| {
+                value.to_str().ok().is_some_and(|val| {
+                    if let Some(addr) = val.strip_prefix("https://") {
+                        addr.starts_with(header_host)
+                    } else {
+                        false
+                    }
```

Comment:
I think this could be a 
```rust
val.strip_prefix("https://").map_or(false, |addr| addr.starts_with(header_host)
```

Reply:
So something like this:
```rust
        let is_https = headers
            .get(http::header::HOST)
            .zip(headers.get(http::header::ORIGIN))
            .and_then(|(host, origin)| host.to_str().ok().zip(origin.to_str().ok()))
            .and_then(|(host, origin)| {
                origin
                    .strip_prefix("https://")
                    .map(|origin| origin.starts_with(host))
            })
            .unwrap_or_default();
```
Also I'd much prefer to construct the cookie already here and not pass a boolean parameter to `Form::insert()`.

Reply:
Adopted your suggestion, looks way nicer. I only changed `unwrap_or_default()` to `unwrap_or(false)`, which is in my view more explicit.

I did however not create the cookie in *src/routes/paste.rs:insert()* cause the `uid` would be needed and that would require additional error handling reworks.

Reply:
Okay, good.